autorest/error.go

```diff
@@ -28,6 +28,9 @@ type DetailedError struct {
 
 	// Message is the error message.
 	Message string
+
+	// Service Error is the response body of failed API.
+	ServiceError interface{}
```

Comment:
why is response body `interface{}`? It doesn't sound like this is actually response body.